astropy/coordinates/tests/test_frames.py

```diff
@@ -543,12 +556,54 @@ def test_represent_as():
     icrs = ICRS(ra=1*u.deg, dec=1*u.deg)
 
     cart1 = icrs.represent_as('cartesian')
-    cart2 = icrs.represent_as(representation.CartesianRepresentation)
+    cart2 = icrs.represent_as(r.CartesianRepresentation)
 
     cart1.x == cart2.x
     cart1.y == cart2.y
     cart1.z == cart2.z
 
+    # now try with velocities
+    icrs = ICRS(ra=0*u.deg, dec=0*u.deg, distance=10*u.kpc,
+                pm_ra_cosdec=0*u.mas/u.yr, pm_dec=0*u.mas/u.yr,
+                radial_velocity=1*u.km/u.s)
+
+    # single string
+    rep2 = icrs.represent_as('cylindrical')
+    assert isinstance(rep2, r.CylindricalRepresentation)
+    assert isinstance(rep2.differentials['s'], r.CylindricalDifferential)
+
+    # single class with positional in_frame_units, verify that warning raised
+    with catch_warnings() as w:
+        icrs.represent_as(r.CylindricalRepresentation, False)
+        assert len(w) == 1
+        assert w[0].category == AstropyWarning
+        assert 'argument position' in str(w[0].message)
+
+    # two classes
+    rep2 = icrs.represent_as(r.CartesianRepresentation,
+                             r.SphericalCosLatDifferential)
```

Comment:
Do we really want a test that checks unwanted behaviour? The reason that this doesn't feel is only because of the work-around for `UnitSphericalRepresentation` and `RadialDifferential`! I'd remove this test and write a `TODO` that things like this should fail in the future (or make them fail already...).